.travis.yml

```diff
@@ -4,44 +4,60 @@ node_js:
     - 10
     - 12
 services:
-  - docker
+    - docker
 env:
-  global:
-    - DEV_BRANCH=dev
-    - RELEASE_BRANCH=main
-    - POST_RELEASE_BRANCH=main
-    - RELEASE_MESSAGE=release
+    global:
+        - DEV_BRANCH=dev
+        - RELEASE_BRANCH=main
+        - POST_RELEASE_BRANCH=main
+        - RELEASE_MESSAGE=release
 cache:
     npm: true
     directories:
-        - "node_modules"
+        - 'node_modules'
 before_script:
     - . ./travis/node-functions.sh
     - VERSION="$(node_load_version)"
     - log_env_variables
 script:
     - npm run build
 jobs:
-  include:
-    - stage: test
-      name: lint
-      script: npm run lint
-    - stage: test
-      name: docker test build
-      script: /bin/bash travis/docker-functions.sh docker_build $VERSION
-      node_js: 10
-    - stage: publish
-      name: docker publish alpha
-      script: /bin/bash travis/docker-functions.sh docker_build $VERSION publish
-      if: branch = env(DEV_BRANCH) AND type = push
-      node_js: 10
-    - stage: release
-      name: docker publish release
-      script: /bin/bash travis/docker-functions.sh docker_build $VERSION release
-      if: branch = env(RELEASE_BRANCH) AND type = api AND commit_message = env(RELEASE_MESSAGE)
-      node_js: 10
-    - stage: post release
-      name: tag and version upgrade
-      script: /bin/bash travis/node-functions.sh node_post_release
-      if: branch = env(RELEASE_BRANCH) AND type = api AND commit_message = env(RELEASE_MESSAGE)
-      node_js: 10
+    include:
+        - stage: test
+          name: lint
+          script: npm run lint
+        - stage: test
+          name: openapi
+          script: npm run test:openapi
+        - stage: test
+          name: openapi typescript client generation
+          script: bash client/generate.sh
```

Comment:
you don't need to repeat "stage" everywhere (especially that stages should be grouped), so something like
```yaml
        - stage: test
          name: lint
          script: npm run lint
        - name: openapi
          script: npm run test:openapi
        - name: openapi typescript client generation
          script: bash client/generate.sh
```

will work in a same way